aea/manager.py

```diff
@@ -0,0 +1,534 @@
+        Return details about agent definition.
+
+        :return: AgentAlias
+        """
+        if agent_name not in self._agents:
+            raise ValueError(f"Agent with name {agent_name} does not exist!")
+        return self._agents[agent_name]
+
+    def _ensure_working_dir(self) -> None:
+        """Create working dir if needed."""
+        if not os.path.exists(self.working_dir):
+            os.makedirs(self.working_dir)
+            self._was_working_dir_created = True
+
+        if not os.path.isdir(self.working_dir):
+            raise ValueError(f"{self.working_dir} is not a directory!")
+        os.makedirs(self._keys_dir)
+
+    def _build_agent_alias(
+        self,
+        project: Project,
+        agent_name: str,
+        agent_overrides=None,
+        component_overrides=None,
+    ) -> AgentAlias:
+        """Create agent alias for project, with given name and overrided values."""
+        json_config = self._make_config(
+            project.path, agent_overrides, component_overrides
+        )
+
+        builder = AEABuilder.from_config_json(json_config, project.path)
+        builder.set_name(agent_name)
+        # builder.set_runtime_mode("async")
+
+        if not builder.private_key_paths:
+            default_ledger = json_config[0].get("default_ledger", DEFAULT_LEDGER)
+            builder.add_private_key(
+                default_ledger, self._create_private_key(agent_name, default_ledger)
+            )
+        agent = builder.build()
+        return AgentAlias(project, agent_name, json_config, agent)
+
+    def _update_dict(self, base: dict, override: dict) -> dict:
+        """Apply overrides for dict."""
+        base = copy.deepcopy(base)
+        base.update(override)
+        return base
+
+    def _make_config(
+        self,
+        project_path: str,
+        agent_overrides: Optional[dict] = None,
+        component_overrides: Optional[List[dict]] = None,
+    ) -> List[dict]:
+        """Make new config baseed on proejct's config with overrides applied."""
+        agent_overrides = agent_overrides or {}
+        component_overrides = component_overrides or []
+
+        if any([key in agent_overrides for key in self.AGENT_DO_NOT_OVERRIDE_VALUES]):
+            raise ValueError(
+                'Do not override any of {" ".join(self.AGENT_DO_NOT_OVERRIDE_VALUES)}'
+            )
+
+        json_data = yaml_load_all(
+            AEABuilder.get_configuration_file_path(project_path).open()
+        )
+        agent_config = self._update_dict(json_data[0], agent_overrides)
+
+        components_configs = {PublicId.from_json(obj): obj for obj in json_data[1:]}
+
+        for obj in component_overrides:
+            component_id = ComponentId(obj["type"], PublicId.from_json(obj))
+            components_configs[component_id] = self._update_dict(
+                components_configs.get(component_id, {}), obj
+            )
+
+        return [agent_config] + list(components_configs.values())
```

Comment:
This needs to be done differently; @marcofavorito will incorporate it in his work next week. You can leave it like this for now.

Reply:
will wait for updates from @marcofavorito